runtime/common/metering.go

```diff
@@ -145,10 +153,17 @@ func NewDictionaryMemoryUsages(length int) (MemoryUsage, MemoryUsage) {
 		}
 }
 
-func NewDictionarySizeUsage(length int) MemoryUsage {
+func NewDictionaryAdditionalSizeUsage(originalSize, additionalSize int) MemoryUsage {
+	var newAmount uint64
+	if originalSize <= 1 {
+		newAmount = uint64(originalSize + additionalSize)
```

Comment:
Does this assume `additionalSize` is always 1? If so, maybe remove the parameter and simplify to `newAmount = 1`. 

If not: What happens if `originalSize <= 1` and `additionalSize > 1`?  Shouldn't this still calculate the new number of intermediate nodes allocated (which won't be the same as the number of new elements)?